clang/test/3C/typedefs.c

```diff
@@ -66,7 +76,18 @@ void barfoo(intptr x) {
 	  *x = 5;
 }
 
+#define MYDECL typedef int* ZZZ;
+MYDECL
+void zzz(void) {
+  int x = 3;
+  ZZZ z = &x;
+}
 
+typedef int * * const a;
+//CHECK: typedef const _Ptr<_Ptr<int>> a;
+void xxx(void) {
+  a b;
```

Comment:
Since this rewrites const (or removes it), you can add a check here, maybe even with a comment on why it's ok to add/remove it.